src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -43,25 +74,63 @@ def __init__(
         self.hub_name = hub_name
         self.region = sagemaker_session.boto_region_name
         self._sagemaker_session = sagemaker_session
+        self._default_thread_pool_size = 20
+        self._s3_client = self._get_s3_client()
+        self.hub_storage_location = self._generate_hub_storage_location(bucket_name)
+
+    def _get_s3_client(self) -> BaseClient:
+        """Returns an S3 client used for creating a HubContentDocument."""
+        return boto3.client("s3", region_name=self.region)
+
+    def _fetch_hub_bucket_name(self) -> str:
+        """Retrieves hub bucket name from Hub config if exists"""
+        try:
+            hub_response = self._sagemaker_session.describe_hub(hub_name=self.hub_name)
+            hub_output_location = hub_response["S3StorageConfig"].get("S3OutputPath")
+            if hub_output_location:
+                location = create_s3_object_reference_from_uri(hub_output_location)
```

Comment:
nit: We should consider logging here so CX knows their `bucket_name` is being overriden